util/topgen/clocks.py

```diff
@@ -101,13 +100,6 @@ def __init__(self,
         raw_clocks = raw.get('clocks', {})
         if not isinstance(raw_clocks, dict):
             raise ValueError(f'clocks for {what} is not a dictionary')
-        for clk_name, src_name in raw_clocks.items():
-            src = sources.get(src_name)
-            if src is None:
-                raise ValueError(f'The {clk_name} entry of clocks for {what} '
-                                 f'has source {src_name}, which is not a '
-                                 f'known clock source.')
-            self.add_clock(clk_name, src)
```

Comment:
If we're removing support for the `clocks` field (great!) then I think we also need to remove lines 100-102 and drop the clock entry in `_asdict`.

Reply:
updated.

As for the `_asdict` portion, they do all clocks that it's being built-up, so i kind of want to keep that.

Reply:
Thanks. For the `_asdict` bit: the problem is that then your code won't properly parse the stuff that comes out of it. Is this motivated by debugging?

Reply:
Yes this is motivated by debug.  The clock field does need to be declared by default in the group, but later we still add clocks into the group themselves and that seems like useful information for debug in the final hjson.

what do you think?

Reply:
Eh, I tend to think that debugging is best done by adding random `print` statements when we come to investigate something :-) But I don't feel very strongly: happy either way.